src/app/clusters/ota-requestor/ota-requestor-server.cpp

```diff
@@ -112,7 +113,9 @@ EmberAfStatus OtaRequestorServerSetUpdateStateProgress(uint8_t value)
         app::DataModel::Nullable<uint8_t> currentValue;
         status = Attributes::UpdateStateProgress::Get(endpoint, currentValue);
         VerifyOrDie(EMBER_ZCL_STATUS_SUCCESS == status);
-        if (currentValue.IsNull() || currentValue.Value() != value)
+
+        if ((currentValue.IsNull() && !value.IsNull()) || (!currentValue.IsNull() && value.IsNull()) ||
+            (!currentValue.IsNull() && !value.IsNull() && (currentValue.Value() != value.Value())))
```

Comment:
Is this just working around Nullable not having an operator==?  How about we just add an `operator==` there, which forwards to the `Optional::operator==`?

Reply:
Do you want me to add `operator==` in this PR or in a separate one?

Reply:
Whichever you prefer.